pages/Group.js

```diff
@@ -57,7 +57,10 @@ function GroupScreen({ navigation }) {
     const { user } = useAuth();
     const [totalBalance, setTotalBalance] = useState();
     const [balances, setBalances] = useState();
-
+    const [offlineMessages, setOfflineMessages] = useState( [] );
```

Comment:
if the user kills the app or moves out of the apps these messages will be cleared
so this implementation will not work since you are storing the offline messages in the memory of the app